src/test/ui/coherence/coherence-negative-outlives-lifetimes.rs

```diff
@@ -1,10 +1,15 @@
+// revisions: stock with_negative_coherence
 #![feature(negative_impls)]
+#![cfg_attr(with_negative_coherence, feature(with_negative_coherence))]
 
 // FIXME: this should compile
 
 trait MyPredicate<'a> {}
-impl<'a, T> !MyPredicate<'a> for &T where T: 'a {}
+
+impl<'a, T> !MyPredicate<'a> for &'a T where T: 'a {}
```

Comment:
`for &'a T` seems like a better representation of the case which should pass, but let me know and I can revert. This ICE happens regardless of it, so it doesn't affect the test for the ICE I'm fixing.